File: app/src/main/java/br/fmu/bioreino/controller/util/Comunicator.java
package br.fmu.bioreino.controller.util;

import android.os.Bundle;
import android.widget.Toolbar;

import androidx.fragment.app.Fragment;

import br.fmu.bioreino.model.entities.Curso;

public interface Comunicador {

    void enviaCurso(Curso curso);

    void enviaCategoria(String categoria);

    void trocaTela(Fragment fragment);

    void trocaTela(Fragment fragment, Bundle bundle);

    void trocaAppBar(Toolbar toolbar);

    void configuraBotaoVoltarSistema(Fragment fragmentAtual);

}

File: app/src/main/java/br/fmu/bioreino/viewModel/util/Comunicator.kt
package br.fmu.bioreino.controller.util

import android.os.Bundle
import android.widget.Toolbar
import androidx.fragment.app.Fragment
import br.fmu.bioreino.model.entities.Curso

interface Comunicator {
    fun enviaCurso(curso: Curso)
    fun enviaCategoria(categoria: String)
    fun changeFragment(fragment: Fragment, bundle: Bundle? = null)
    fun trocaAppBar(toolbar: Toolbar)
    fun configuraBotaoVoltarSistema(fragmentAtual: Fragment)
}